### PR TITLE
Chore: upgrade eslint-plugin-eslint-plugin

### PR DESCRIPTION
**Pull request by aladdin-add**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ x] Other, please explain:

#### What changes did you make? (Give an overview)
upgrade eslint-plugin-eslint-plugin to latest.

#### Is there anything you'd like reviewers to focus on?
it also rewrite some generators to remove disable directives.

### CODE REVIEW

**Review by samarmohan — APPROVED** [review]

File: lib/rules/dot-notation.js [hunk 1]
```diff
@@ -87,36 +87,38 @@ module.exports = {
...
+                    fix(fixer) {
+                        const fixes = [];
                         const leftBracket = sourceCode.getTokenAfter(node.object, astUtils.isOpeningBracketToken);
                         const rightBracket = sourceCode.getLastToken(node);
                         const nextToken = sourceCode.getTokenAfter(node);
 
                         // Don't perform any fixes if there are comments inside the brackets.
                         if (sourceCode.commentsExistBetween(leftBracket, rightBracket)) {
-                            return; // eslint-disable-line eslint-plugin/fixer-return -- false positive
+                            return null;
```
Comment (mdjermanovic) [inline lib/rules/dot-notation.js:98]: Isn't this a bug that should be fixed in `eslint-plugin-eslint-plugin`?

Comment (aladdin-add) [reply lib/rules/dot-notation.js:98]: created an issue https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/issues/117 to track it.

Comment (aladdin-add) [reply lib/rules/dot-notation.js]: reverted the changes; let's make another PR when it is fixed in the plugin.

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Pull request merged**